graviti/utility/common.py

```diff
@@ -67,3 +68,48 @@ def shorten(origin: str) -> str:
 
     """
     return origin[:7]
+
+
+class CheckArgs:
+    """Check the input arguments of a function.
+
+    Arguments:
+        args: The names of the arguments to be checked.
+
+    """
+
+    def __init__(self, *args: str) -> None:
+        self._names = args
+
+    def __call__(self, func: Callable[..., Any]) -> Callable[..., Any]:
+        """Check the arguments type of the function.
+
+        Arguments:
+            func: The function whose arguments to check.
+
+        Returns:
+            The function added arguments checking code.
+
+        """
+        signature = inspect.signature(func)
+        parameters = signature.parameters
+
+        @wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            arguments = signature.bind(*args, **kwargs).arguments
+            for name in self._names:
+                type_ = parameters[name].annotation
```

Comment:
what if the `name` is an optional parameter?

Reply:
Optional will be expressed as Union in annotations.